src/ArmTemplates/Creator/TemplateCreators/TagTemplateCreator.cs

```diff
@@ -24,51 +26,64 @@ public TagTemplateCreator(ITemplateBuilder templateBuilder)
 
         public Template CreateTagTemplate(CreatorParameters creatorConfig)
         {
-            // create empty template
-            Template tagTemplate = this.templateBuilder.GenerateEmptyTemplate().Build();
+            var tagTemplate = this.templateBuilder.GenerateTemplateWithApimServiceNameProperty().Build();
+            var tagsDictionary = new Dictionary<string, string>();
 
-            // add parameters
-            tagTemplate.Parameters = new Dictionary<string, TemplateParameterProperties>
+            void AddTagNameToDictionary(string tagName, Dictionary<string, string> tagsDictionary)
             {
-                {ParameterNames.ApimServiceName, new TemplateParameterProperties(){ Type = "string" }}
-            };
+                var resourceName = ResourceNamingHelper.GenerateValidResourceNameFromDisplayName(tagName);
 
-            // aggregate all tags from apis
-            HashSet<string> tagHashset = new HashSet<string>();
-            List<ApiConfig> apis = creatorConfig.Apis;
-            if (apis != null)
+                if (tagsDictionary.ContainsKey(resourceName))
+                {
+                    var existingValue = tagsDictionary[resourceName];
+                    if (!existingValue.Equals(tagName))
+                    {
+                        throw new DuplicateTagResourceNameException(string.Format(ErrorMessages.DuplicateTagResourceNameErrorMessage, existingValue, tagName, resourceName));
+                    }
+                }
+                else
+                {
+                    tagsDictionary.Add(resourceName, tagName);
+                }
+            }
+
+            if (!creatorConfig.Apis.IsNullOrEmpty())
             {
-                foreach (ApiConfig api in apis)
+                foreach (var api in creatorConfig.Apis)
                 {
-                    if (api.Tags != null)
+                    if (!api.Tags.IsNullOrEmpty())
                     {
-                        string[] apiTags = api.Tags.Split(", ");
-                        foreach (string apiTag in apiTags)
+                        var apiTags = api.Tags.Split(", ");
```

Comment:
Please, add a corner case test for this code like
`apiTags = mytag1,mytag2, mytag3, `

Reply:
I guess 3 tags have to be parsed - myTag1, myTag2 and myTag3

Reply:
`apiTags = mytag1,mytag2, mytag3,` This example in PR changes will raise an Exception with EmptyResourceName message